src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -529,4 +1045,110 @@ mod test {
+            let value = ValueType::String("test_watch_value2".to_owned());
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            client_clone.set(&key, &value, None).await.unwrap();
+        });
+        let key = KeyType::String("test_watch".to_owned());
+        let value = ValueType::String("test_watch_value2".to_owned());
+        let watch_result = watch_stream.message().await.unwrap().unwrap();
+        assert_eq!(watch_result.0, key.to_string_key());
+        assert_eq!(watch_result.1.unwrap(), value);
+
+        // Delete a key
+        let client_clone = client.clone();
+        tokio::spawn(async move {
+            let key = KeyType::String("test_watch".to_owned());
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            client_clone.delete(&key, None).await.unwrap();
+        });
+        let key = KeyType::String("test_watch".to_owned());
+        client.delete(&key, None).await.unwrap();
+        let watch_result = watch_stream.message().await.unwrap();
+        assert!(watch_result.is_none());
+
+        // Close stream
+        let res = watch_stream.cancel().await.unwrap();
+        assert_eq!(res, ());
+    }
+
+    #[tokio::test]
+    async fn test_create_session() {
+        let client = EtcdKVEngine::new_for_local_test(vec![ETCD_ADDRESS.to_owned()])
+            .await
+            .unwrap();
+        let lease_id = client.create_lease(5).await.unwrap();
+        // Set a temp key for this lease
+        let key = KeyType::String("test_create_session".to_owned());
+        let value = ValueType::String("test_create_session_value".to_owned());
+        client.set(&key, &value, Some(SetOption::new().with_lease(lease_id)))
+            .await
+            .unwrap();
+
+        assert_eq!(client.get(&key).await.unwrap().unwrap(), value);
+
+        // Test keep alive
+        let session = client.create_session(lease_id, 5).await.unwrap();
+        tokio::time::sleep(Duration::from_secs(10)).await;
+        assert_eq!(client.get(&key).await.unwrap().unwrap(), value);
+
+        // Test cancel keepalive
+        drop(session);
+        tokio::time::sleep(Duration::from_secs(10)).await;
+        assert!(client.get(&key).await.unwrap().is_none());
+    }
+
+    #[tokio::test]
+    async fn test_campaign() {
+        let client = EtcdKVEngine::new_for_local_test(vec![ETCD_ADDRESS.to_owned()])
+            .await
+            .unwrap();
+        let txn = client.new_meta_txn().await;
+
+        let key = KeyType::String("test_campaign".to_owned());
+        let val = "test_campaign_val".to_owned();
+        let val2 = "test_campaign_val2".to_owned();
+        let lease_id = client.create_lease(5).await.unwrap();
+
+        // Campaign the key
+        let (campaign_status, campaign_val) = txn.campaign(&key, val.clone(), lease_id).await.unwrap();
+        assert_eq!(campaign_status, true);
+        assert_eq!(campaign_val, val);
+
```

Comment:
2nd time: Should use a different session/lease_id. Expect campain to overwrite the value,  and then drop the original session.
Maybe test query the key and confirm new lease_id.

Reply:
Thanks, I've updated with new `lease_id` before each `campaign`.

BTW, I've add a `get_lease_id` for test to get lease_id from `get`.

Reply:
Should only use session, lease is a low level api